src/components/ChannelPreview/ChannelPreview.js

```diff
@@ -4,26 +4,30 @@ import PropTypes from 'prop-types';
 import { ChatContext } from '../../context';
 import { useLatestMessagePreview } from './hooks/useLatestMessagePreview';
 
-/**
- * NOTE: We created an inner component which only receives `client` from the ChatContext. This
- * practice will prevent unnecessary renders of the component when items in ChatContext change
- */
-const ChannelPreviewWithContext = React.memo((props) => {
-  const { channel, client } = props;
+const ChannelPreview = (props) => {
+  const { channel } = props;
+  const { client } = useContext(ChatContext);
   const [lastMessage, setLastMessage] = useState({});
   const [unread, setUnread] = useState(channel.countUnread());
   const latestMessage = useLatestMessagePreview(channel, lastMessage);
 
   useEffect(() => {
-    const handleNewMessageEvent = (e) => {
+    const handleEvent = (e) => {
       setLastMessage(e.message);
-      setUnread(channel.countUnread());
+
+      if (e.type === 'message.new') {
```

Comment:
couldn't message.deleted technically change the unread count?

Reply:
With our current js client implementation, it won't!